inference-engine/tests/functional/plugin/cpu/single_layer_tests/roi_pooling.cpp

```diff
@@ -0,0 +1,162 @@
+                CommonTestUtils::fill_data_roi(blob->buffer(), blob->size(), feat_map_shape[0] - 1, height, width, 1.0f, is_roi_max_mode);
+            } else {
+                blob = GenerateInput(*info);
+            }
+            inputs.push_back(blob);
+            it++;
+        }
+    }
+
+    void SetUp() {
+        LayerTestsDefinitions::roiPoolingParamsTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        InferenceEngine::SizeVector inputShape;
+        InferenceEngine::SizeVector coordsShape;
+        InferenceEngine::SizeVector poolShape;
+        InferenceEngine::Precision netPrecision;
+
+        // threshold = 0.08f;
+
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        std::tie(inputShape, coordsShape, poolShape, spatial_scale, pool_method, netPrecision, targetDevice) = basicParamsSet;
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputShape, coordsShape});
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+        std::shared_ptr<ngraph::Node> roi_pooling = ngraph::builder::makeROIPooling(paramOuts[0], paramOuts[1], poolShape, spatial_scale, pool_method);
+        ngraph::ResultVector results{std::make_shared<ngraph::opset3::Result>(roi_pooling)};
+
+        function = makeNgraphFunction(ngPrc, params, roi_pooling, "roi_pooling");
+
+        selectedType = getPrimitiveType() + "_FP32";
+    }
+
+private:
+    ngraph::helpers::ROIPoolingTypes pool_method;
+    float spatial_scale;
+};
+
+TEST_P(ROIPoolingCPULayerTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    Run();
+    CheckPluginRelatedResults(executableNetwork, "ROIPooling");
+}
+
+namespace {
+
+std::vector<CPUSpecificParams> filterCPUInfoForDevice() {
+    std::vector<CPUSpecificParams> resCPUParams;
+    if (with_cpu_x86_avx512f()) {
+        resCPUParams.push_back(CPUSpecificParams{{nChw16c, nc}, {nChw16c}, {"jit_avx512"}, "jit_avx512"});
+    } else if (with_cpu_x86_avx2()) {
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc}, {nChw8c}, {"jit_avx2"}, "jit_avx2"});
+    } else if (with_cpu_x86_sse42()) {
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc}, {nChw8c}, {"jit_sse42"}, "jit_sse42"});
+    } else {
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc}, {nChw8c}, {"ref"}, "ref"});
+    }
+    return resCPUParams;
+}
+
+const std::vector<std::vector<size_t>> inShapes = {{1, 3, 8, 8}, {3, 4, 50, 50}};
+
+const std::vector<std::vector<size_t>> pooledShapes_max = {{1, 1}, {2, 2}, {3, 3}, {6, 6}};
+
+const std::vector<std::vector<size_t>> pooledShapes_bilinear = {{1, 1}, {2, 2}, {3, 3}, {6, 6}};
+
+const std::vector<std::vector<size_t>> coordShapes = {{1, 5}};
+                                                      // {3, 5},
```

Comment:
For some reason those shapes are not suited for roi_pooling in case of BF16.
Under investigation.

Reply:
The reason was in input generation stage.
Input blob was filled with float offsets, which does not work for bfloat